services/authService.js

```diff
@@ -37,18 +66,23 @@ export const authorizeUser = (integrationUrl) => {
     })
     .catch((error) => {
       const response = error.response || {};
-      if (response.status === 401) {
+      if (response.status == 401) {
         const userSession = response.data || {};
         const properties = userSession.properties || {};
         if (properties.authorizationUrl != undefined) {
-          openAuthorizationPopupWindow(properties.authorizationUrl);
-
-          return Promise.resolve({
-            success: false,
-          });
+          if (openAuthorizationPopupWindow(properties.authorizationUrl)) {
+            // we are polling this until we get a 200 or reach MAX_NUM_OF_ATTEMPTS
+            return pollUserInfo(integrationUrl, token).then((success) => {
```

Comment:
You could resolve the promise on pollUserInfo adding the token, and here just return the promisse: 
return pollUserInfo(integrationUrl, token).then((result) => { return result } );

Reply:
Done (I guess). Check it out please.